File: Python/dsaOne/Javascript/Arrays_and_Hashing/twoSum.js
//https://leetcode.com/problems/two-sum/description/

const twoSum = (nums, target) => {
    var map = new Map()
    for(let i = 0; i < nums.length; i++){
        var num = nums[i]
        if (map.get(target-num) == undefined) map.set(num, i)
        else return [i, map.get(target-num)].sort()
    }
}

console.log(twoSum([1,2,3,4], 7))

File: Python/dsaOne/Blind 75/arrays_and_hashing./twoSum.py
#https://leetcode.com/problems/two-sum/

def twoSum(nums, target):
    hm = {}
    for i in range(len(nums)):
        another = target - nums[i]
        if another in hm:
            return [i, hm[another]]
        else:
            hm[nums[i]] = i
    return []

print(twoSum([2,4,6,1], 10))
print(twoSum([2,4,6,1], 17))
